service/controller/v15/encrypter/vault/vault.go

```diff
@@ -197,6 +201,78 @@ func (e *Encrypter) DesiredState(ctx context.Context, customObject v1alpha1.AWSC
 	return state, nil
 }
 
+func (e *Encrypter) EnsureCreatedAuthorizedIAMRoles(ctx context.Context, iamRoleARNs ...string) error {
+	e.logger.LogCtx(ctx, "level", "debug", "message", "ensuring authorized vault IAM roles in the Vault API")
+
+	err := e.ensureToken()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	p := path.Join("auth", "aws", "role", decrypterVaultRole)
+
+	role, err := e.getAWSAuthRole(p)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	combinedIAMRoleARNs := stringSlice(role.BoundIAMRoleARN).Add(iamRoleARNs...)
+
+	if len(combinedIAMRoleARNs) != len(role.BoundIAMRoleARN) {
```

Comment:
One improvement here could be to have `stringSlice.Add` returning the number of added elements (so modifying the object in place instead of returning a new one).
And then just check if number of added elements is greater than 1.
No blocker just a suggestion.